### PR TITLE
Tailscale fork rebased on top of v0.0.20201118

### PR DESCRIPTION
**Pull request by peske**
Our `main` branch rebased on top of [upstream v0.0.20201118](https://git.zx2c4.com/wireguard-go/tag/?h=v0.0.20201118) (the latest tag at this moment).

### CODE REVIEW

**Comment by peske** [comment]
@bradfitz , the only commit from our fork that I've skipped in this rebase is yours: https://github.com/tailscale/wireguard-go/commit/2c83f43a98595a46bd039df308e79d93679bf5be

It looks to me that it isn't applicable anymore. But can you please take a look at `tun/tun_darving.go` (file affected by the commit), to ensure that it's OK?

Thanks!

**Comment by bradfitz** [comment]
> @bradfitz , the only commit from our fork that I've skipped in this rebase is yours: 2c83f43

Yes, that's been fixed in two different ways upstream: both in wireguard-go and x/sys/unix. So you can ignore that change.

**Comment by peske** [comment]
I've reviewed carefully all the conflicts I've had, and I believe that I've resolved them correctly. But if you want to recheck, this is the list of our commits that caused conflicts:
```
737408978d5ba867ff44e5066fdbdfe97b7a1c8c
a1fbb1c4ad4d769da533a08bed29e30656b04e75
f44d4e60f3fbbe2a21512e062486eb0a77c03e56 device/peer.go
abfd140f038db15c9d113c6dcf5e83e4174ff01d
490a399e24e71d1dcddc743bbd5a0397a71ae5b9 tun/wintun/wintun_windows.go
2996be34502d014f9375aa96210e110534acbfa0
07f64e3605c8b96b2821c08093a66136bd8c4cb4 (file deleted)
2c83f43a98595a46bd039df308e79d93679bf5be (I've discarded this commit because it isn't applicable anymore.)
a6168fd06b3f3b5608b2753e24f8c77307dce64f (file deleted)
```

File: go.mod [hunk 1]
```diff
@@ -6,4 +6,5 @@ require (
 	golang.org/x/crypto v0.0.0-20201117144127-c1f2f97bffc9
 	golang.org/x/net v0.0.0-20201110031124-69a78807bb2b
 	golang.org/x/sys v0.0.0-20201117222635-ba5294a509c7
+	golang.org/x/text v0.3.3
```
Comment (josharian) [inline go.mod:9]: I don't see where this is used.

Comment (peske) [reply go.mod]: You're right - it isn't used. Not sure when and why it appeared there - probably during some build attempt while testing rebase. I've just removed it.

Comment (josharian) [reply go.mod]: Thanks. In an ideal world, you'd remove this commit instead of adding a new one (partially) reverting it.

**Comment by josharian** [comment]
Once the dependency-related commits have been removed, this LGTM.

For the record, I reviewed by diffing two diffs, namely each of the branches vs its upstream. There were only a few different differences, all readily accountable for.

Thanks for doing this. Do you have permissions to push this as the new main branch, or do you need me to?

**Comment by peske** [comment]
Deleted the wrong commits as suggested, and pushed directly to `main`. Closing this PR.

**Pull request closed without merging**